tao_compiler/mlir/disc/transforms/disc_to_llvm.cc

```diff
@@ -1329,6 +1329,35 @@ LogicalResult PrintfToLLVMPattern::matchAndRewrite(
   return success();
 }
 
+// Converts a ral.dispatch_op to its llvm format.
+struct GetPointerToLLVMPattern : public ConvertOpToLLVMPattern<GetPointerOp> {
+  GetPointerToLLVMPattern(LLVMTypeConverter& type_converter,
+                          SymbolTable& symbol_table)
+      : ConvertOpToLLVMPattern<GetPointerOp>(type_converter),
+        symbol_table_(symbol_table) {}
+
+  LogicalResult matchAndRewrite(
+      GetPointerOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter& rewriter) const override;
+
+ private:
+  SymbolTable& symbol_table_;
+};
+
+LogicalResult GetPointerToLLVMPattern::matchAndRewrite(
+    GetPointerOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter& rewriter) const {
+  auto input = adaptor.getOperands()[0];
+  auto loc = op.getLoc();
+  MemRefDescriptor memref(input);
+
+  Value allocated_bytes_ptr = rewriter.create<LLVM::PtrToIntOp>(
+      loc, getIndexType(), memref.allocatedPtr(rewriter, loc));
+  rewriter.replaceOpWithNewOp<LLVM::BitcastOp>(op, rewriter.getIntegerType(64),
```

Comment:
why we use int64 to represent pointer ?

Reply:
i64 can represent large address than i8/i32.